central/complianceoperator/v2/report/manager/watcher/scanwatcher_test.go

```diff
@@ -0,0 +1,324 @@
+			Annotations: map[string]string{checkCountAnnotationKey: checkCount},
+		})
+		require.NoError(t, err)
+	}
+}
+
+func handleResult(id string) func(*testing.T, ScanWatcher) {
+	return func(t *testing.T, scanWatcher ScanWatcher) {
+		err := scanWatcher.PushCheckResult(&storage.ComplianceOperatorCheckResultV2{
+			CheckId: id,
+		})
+		require.NoError(t, err)
+	}
+}
+
+func TestScanWatcher(t *testing.T) {
+	cases := map[string]struct {
+		events          []testEvent
+		assertScanID    string
+		assertResultIDs []string
+	}{
+		"scan ready -> result -> result": {
+			events: []testEvent{
+				handleScanWithAnnotation("id-1", "2"),
+				handleResult("id-1"),
+				handleResult("id-2"),
+			},
+			assertScanID:    "id-1",
+			assertResultIDs: []string{"id-1", "id-2"},
+		},
+		"scan -> result -> result -> scan ready": {
+			events: []testEvent{
+				handleScan("id-1"),
+				handleResult("id-1"),
+				handleResult("id-2"),
+				handleScanWithAnnotation("id-1", "2"),
+			},
+			assertScanID:    "id-1",
+			assertResultIDs: []string{"id-1", "id-2"},
+		},
+		"scan -> result -> scan ready -> result": {
+			events: []testEvent{
+				handleScan("id-1"),
+				handleResult("id-1"),
+				handleScanWithAnnotation("id-1", "2"),
+				handleResult("id-2"),
+			},
+			assertScanID:    "id-1",
+			assertResultIDs: []string{"id-1", "id-2"},
+		},
+		"result -> result -> scan ready": {
+			events: []testEvent{
+				handleResult("id-1"),
+				handleResult("id-2"),
+				handleScanWithAnnotation("id-1", "2"),
+			},
+			assertScanID:    "id-1",
+			assertResultIDs: []string{"id-1", "id-2"},
+		},
+	}
+	for tName, tCase := range cases {
+		t.Run(tName, func(t *testing.T) {
+			watcherID := "id"
+			ctx, cancel := context.WithCancel(context.Background())
+			defer cancel()
+			resultQueue := queue.NewQueue[*ScanWatcherResults]()
+			scanWatcher := NewScanWatcher(ctx, watcherID, resultQueue)
+			for _, event := range tCase.events {
+				event(t, scanWatcher)
+			}
+			require.Eventually(t, func() bool {
+				return resultQueue.Len() > 0
+			}, 200*time.Millisecond, 10*time.Millisecond)
+			result := resultQueue.Pull()
+			require.NotNil(t, result)
+			assert.Equal(t, tCase.assertScanID, result.Scan.GetId())
+			for _, checkID := range tCase.assertResultIDs {
+				found := false
+				for checkResult := range result.CheckResults {
+					if checkID == checkResult {
+						found = true
+						break
+					}
+				}
+				assert.Equal(t, true, found)
+			}
+		})
+	}
+}
+
+func TestScanWatcherCancel(t *testing.T) {
+	watcherID := "id"
+	ctx, cancel := context.WithCancel(context.Background())
+	resultQueue := queue.NewQueue[*ScanWatcherResults]()
+	scanWatcher := NewScanWatcher(ctx, watcherID, resultQueue)
+	handleScan("id-1")(t, scanWatcher)
+	handleResult("id-1")(t, scanWatcher)
+	cancel()
+	select {
+	case <-scanWatcher.Finished().Done():
+	case <-time.After(100 * time.Millisecond):
+		t.Error("timeout waiting for the watcher to stop")
+	}
+	assert.Equal(t, 0, resultQueue.Len())
```

Comment:
Optionally
```suggestion
	assert.Len(t, resultQueue, 0)
```

Reply:
Naming: resultQueue vs `readyQueue` in the code. Maybe we could stick to `readyQueue`?